### PR TITLE
more traditional feature flag system

### PR DESCRIPTION
**Pull request by maxtaco**
- ask the server if it's live or not, and cache it for as long as the server says

### CODE REVIEW

**Comment by maxtaco** [comment]
quick review on this, it's pretty simple

**Comment by maxtaco** [comment]
cc @heronhaye, this system I prefer....

**Comment by heronhaye** [comment]
lgtm, is there server support as well?

**Comment by maxtaco** [comment]
No, just a flag to not send down signatures, which was in there for a while.

File: go/libkb/features.go [hunk 1]
```diff
@@ -33,3 +35,109 @@ func (set FeatureFlags) Admin() bool {
 func (set FeatureFlags) Empty() bool {
 	return len(set) == 0
 }
+
+type featureSlot struct {
+	sync.Mutex
```
Comment (mlsteele) [inline go/libkb/features.go]: Wouldn't locktab be better?

Comment (maxtaco) [reply go/libkb/features.go]: this seemed simpler. i can change it

**Comment by patrickxb** [comment]
approved

**Pull request merged**